weldx/geometry.py

```diff
@@ -99,255 +98,404 @@ def _to_list(var) -> list:
+
+    def _get_component_derivative_squared(self, i: int) -> sympy.Expr:
+        """Get the derivative of an expression for the i-th vector component."""
+
+        def _get_component(v, i):
+            if isinstance(v, Q_):
+                v = v.to_base_units().m
+            if v.size == 3:
+                return v[i]
+            return float(v)
+
+        me = self._series.data
+        subs = [(k, _get_component(v.data, i)) for k, v in me.parameters.items()]
+
+        sym = sympy.sympify(self._series.position_dim_name)
+        return me.expression.subs(subs).diff(sym) ** 2
+
+    def _get_length_expr(self) -> MathematicalExpression:
+        """Get the primitive of a the trace function if it is expression based."""
+        der_sq = [self._get_component_derivative_squared(i) for i in range(3)]
+        expr = sympy.sqrt(der_sq[0] + der_sq[1] + der_sq[2])
+        mc, u = sympy.symbols("max_coord, unit")
+        primitive = sympy.integrate(expr, (self._series.position_dim_name, 0, mc)) * u
+        params = dict(unit=Q_(1, Q_("1mm").to_base_units().u).to(_DEFAULT_LEN_UNIT))
+
+        return MathematicalExpression(primitive, params)
+
+    def _len_section_disc(self, position: float) -> pint.Quantity:
+        """Get the length until a specific position on the trace (discrete version)."""
+        if position >= self._max_coord:
+            diff = self._series.data[1:] - self._series.data[:-1]
+        else:
+            pdn = self._series.position_dim_name
+            coords = self._series.coordinates[pdn].data
+            idx_coord_upper = np.abs(coords - position).argmin()
+            if coords[idx_coord_upper] < position:
+                idx_coord_upper = idx_coord_upper + 1
+
+            coords_eval = np.append(coords[:idx_coord_upper], position)
+            vecs = self._series.evaluate(**{pdn: coords_eval}).data
+
+            diff = vecs[1:] - vecs[:-1]
+
+        length = np.sum(np.linalg.norm(diff.m, axis=1))
+        return Q_(length, diff.u)
+
+    def get_section_length(self, position: float) -> pint.Quantity:
+        """Get the length from the start of the segment to the passed relative position.
 
         Parameters
         ----------
-        point_start :
-            Starting point of the segment
-        point_end :
-            End point of the segment
+        position:
+            The value of the relative position coordinate.
 
         Returns
         -------
-        LineSegment
-            Line segment
+        pint.Quantity:
+            The length at the specified value.
 
         """
-        points = np.transpose(np.array([point_start.m, point_end.m], dtype=float))
-        return cls(Q_(points, _DEFAULT_LEN_UNIT))
+        if self._series.is_expression:
+            length = self._length_expr.evaluate(max_coord=position).data
+        else:
+            length = self._len_section_disc(position=position)
+        if length.m <= 0:
```

Comment:
```suggestion
        if length <= 0:
```

Reply:
Shouldn't that trigger a dimensionality error? Well, let's try.